src/thread_pool_async.mli

```diff
@@ -1,22 +1,94 @@
 open Async_kernel
 
+(** A thread pool. *)
 type 'state t
 
-type 'result computation =
-  [ `Ok of 'result
-  | `Attempt_retry ]
+(** A single worker thread within the pool. *)
+type 'state worker
 
+(** Raised if the pool is initiated with a non-positive thread count. *)
+exception Invalid_thread_count of int
```

Comment:
I'm wondering, since the number can be checked in the initialization would it not make more sense to avoid the exception and make initialization return a result?

Reply:
Well, I don't know, I don't think it really helps anything in practice. I guess that if a thread failed to get reserved for the pool, things would blow up in a terrible way anyway. This is sort of a similar condition in my view.

If you care about this a lot, then make the change.

Reply:
Yeah, let's roll with the code as-is for now, especially as you said that this change breaks users.